--- /home/umbi/Locali/Temp/bme690/bsec_v3-1-0-0/examples/BSEC_Integration_Examples/examples/bsec_iot_example/bsec_integration.h
+++ /home/umbi/ProgettiESP/POLVERINE_GITHUB/Polverine/POLVERINE_FULL_MQTT_DEMO/src/bsec_integration.h
@@ -77,7 +77,7 @@
 #include "bsec_interface.h"
 #include "bsec_datatypes.h"
 
-#define NUM_OF_SENS    	                UINT8_C(8)
+#define NUM_OF_SENS    	                1 //UINT8_C(8)
 #define BSEC_INSTANCE_SIZE              UINT16_C(3272)
 #define BSEC_CHECK_INPUT(x, shift)		(x & (1 << (shift-1)))
 #define BSEC_TOTAL_HEAT_DUR             UINT16_C(140)
@@ -110,8 +110,10 @@
  *	The default offset provided has been determined by testing the sensor in LP and ULP mode on application board 3.0
  *	Please update the offset value after testing this on your product 
  */
-#define TEMP_OFFSET_LP		(0.1495f)
-#define TEMP_OFFSET_ULP		(0.466f)
+//#define TEMP_OFFSET_LP		(0.1495f)
+//#define TEMP_OFFSET_ULP		(0.466f)
+#define TEMP_OFFSET_LP		(6.1495f)
+#define TEMP_OFFSET_ULP		(6.466f)
 
 
 /* Structure to store the BSEC output values */
@@ -154,7 +156,7 @@
 typedef void (*bme69x_interface_fct)(struct bme69x_dev *bme, uint8_t intf, uint8_t sen_no);
 
 /* function pointer to the function processing obtained BSEC outputs */
-typedef void (*output_ready_fct)(char *output);
+typedef void (*output_ready_fct)(outputs_t *output);
 
 /* function pointer to the function saving BSEC state to NVM */
 typedef void (*state_save_fct)(const uint8_t *state_buffer, uint32_t length);

